src/shared/options/readOptions.ts

```diff
@@ -238,3 +243,37 @@ export async function readOptions(
 		options: augmentedOptions,
 	};
 }
+
+async function getBase(): Promise<OptionsBase> {
```

Comment:
[Testing] Lots of great logic here, but no unit tests. This function is something I think should be tested. It's got some tricky logic.

Suggestion: it might be easiest to test it if you extract `getBase` into a new file?